packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx

```diff
@@ -0,0 +1,45 @@
+import { createElement, ReactNode, ReactElement } from "react";
+import { ObjectItem } from "mendix";
+import Swiper, { SwiperOptions, Navigation, Pagination, EffectFade, Autoplay } from "swiper";
+import { Swiper as SwiperReact, SwiperSlide } from "swiper/react";
+
+export interface CarouselProps<T extends ObjectItem> {
+    pagination: boolean;
+    loop: boolean;
+    tabIndex?: number | undefined;
+    animation?: boolean;
+    autoplay?: boolean;
+    delay?: number;
+    navigation: boolean;
+    onClick: () => void;
+    items: T[];
+    itemRenderer: (renderWrapper: (children: ReactNode) => ReactElement, item: T) => ReactNode;
+}
+
+export function Carousel<T extends ObjectItem>(props: CarouselProps<T>): ReactElement {
+    const { items, pagination, loop, tabIndex, animation, autoplay, delay, navigation, itemRenderer, onClick } = props;
+    Swiper.use([Navigation, Pagination, EffectFade, Autoplay]);
```

Comment:
- Is there no other method to inject these dependencies that is more React-lifecycle friendly? 
- If not, does this need to be called on every cycle? If not, let's wrap it in a `useEffect(..., []);`

Reply:
- no, React-lifecycle friendly methods introduced recently (1 of Feb), but because of bundling we can't use new version now.
- There is only one cycle happening, so basically it renders only once. That's why I didn't put it in a `useEffect` hook

Reply:
> There is only one cycle happening,  so basically it renders only once

If you think that's a realistic assumption (after fixing the loading state stuff), then fine to keep it like this. But with `useEffect` we know for sure that this call will happen only once 😄

Reply:
Agree, but as you  were always saying, avoid unnecessary calls :)

Reply:
But I think in this case it will prevent to execute more than once as it will execute only when the component mounted.

```
useEffect(() =>  Swiper.use([Navigation, Pagination, EffectFade, Autoplay]), []);
```

Reply:
☝️  Here you're _always_ calling it, while with `useEffect` you're only calling it once per React lifecycle